### ISSUE TITLE
(at least areg): optionally do not abbreviate account names

### ISSUE THREAD

**Issue by jmtd** [simonmichael/hledger#1995]
The areg documentation says

```-   the names of all the other account(s) involved in this transaction
    (probably abbreviated)```

In my experience my account names are always abbreviated - perhaps that's a function of their length, it does not seem to be influenced by the width for default output format.

It would be great if areg -- and any other command which abbreviates command names -- could be told not to.

**Comment by simonmichael** [comment]
Thanks. I believe aregister is the only command which abbreviates always, regardless of width; it just assumes it will need to. I agree some smarter behaviour would be nice. I guess you are willing to use a very wide terminal, or panning with a pager or similar, to view unelided wide reports.

**Comment by jonestristand** [comment]
Agree, would love to see this fixed. It's more tricky and error prone to get the report I want from register, but for sending stuff to my accountant, having the full account name is essential. Any progress?

**Comment by simonmichael** [comment]
No progress. Is there anyone who'd like to look at this ? It involves looking at how register does it in Register.hs and porting some of that to Aregister.hs.

**Comment by savanto** [comment]
I encountered this issue recently, specifically with `areg`'s CSV output, which also always had abbreviated account names. I don't mind the abbreviations when outputting to text/terminal, but I do post-processing on the CSV and then supply it to an accountant, and I needed the account names to be unabridged. I was working around this by building a mapping of `abbrev->full` account names in my post-processing tool, but it seems to me that machine-readable output should not be applying an abbreviation strategy at all, and leaving decisions like that to the post-processor. 

I implemented a small fix to this (sub-)problem in #2416 by simply not abbreviating accounts in CSV/HTML/FODS output, while leaving text output unchanged.

Since I have this piece of the codebase in my head right now, I'll note that `aregister` uses [`accountSummarisedName`](https://github.com/simonmichael/hledger/blob/5d8b30ba371b03dbf08054e88a2bd1673865c74e/hledger-lib/Hledger/Data/AccountName.hs#L146) to trim all account components but the last one down to two characters no matter what, which is a pretty blunt approach. In contrast `register` appears to use [`elideAccountName`](https://github.com/simonmichael/hledger/blob/5d8b30ba371b03dbf08054e88a2bd1673865c74e/hledger-lib/Hledger/Data/AccountName.hs#L318) which not only uses a width parameter to determine whether to abbreviate at all, but decides whether to abbreviate or elide entirely.

I didn't use this "smart" strategy for `aregister` in #2416 to limit the scope of the change I was making. However I do think it might be reasonable to make `aregister` work the same way as `register` by swapping `accountSummarisedName` for `elideAccountName` wherever its still being used in non-machine output. Depends on whether width information is already readily available. One additional challenge is that it's not enough to simply abbreviate account names to fit the given width, because it's the `T.intercalate ", " acctnames` is what needs to fit the width for `areg`, which is slightly harder.

Lastly, I'm content with text/terminal `areg` accounts always being abridged, as long as CSV output is unabridged. So, I don't see a need for a command line flag to control this. But I'm open to implementing such a flag if it's useful (vs the tradeoff of having yet another option), with input on what exactly this flag controls (whether to abbreviate? how to abbreviate? which mode to abbreviate? etc).

**Comment by simonmichael** [comment]
@jmtd: with #2416 merged, areg now abbreviates names only in terminal (txt) output. In HTML or CSV output eg, they'll be unabbreviated. 

Does this solve it, or are you still wishing for full account names in the txt output ?

**Comment by simonmichael** [comment]
As far as I know, aregister is the only report that abbreviates account names to two-letter parts.

Closing for now.
